### PR TITLE
feat(helm): add podmonitor

### PR DESCRIPTION
**Pull request by aslafy-z**
Add support for podMonitor in helm chart.

Testing done:
- podmonitor generated
image

- prometheus discovered it
image

### CODE REVIEW

File: deploy/manifests/controller/helm/retina/templates/daemonset.yaml [hunk 1]
```diff
@@ -60,7 +60,8 @@ spec:
           {{- end}}
           {{- end}}
           ports:
-          - containerPort: {{ .Values.retinaPort }}
+          - name: http
```
Comment (vakalapa) [inline deploy/manifests/controller/helm/retina/templates/daemonset.yaml:63]: can we make this name retina ? so it is specific to this daemonset,

File: deploy/manifests/controller/helm/retina/templates/podmonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  name: {{ include "retina.fullname" . }}
+  namespace: {{ ternary .Values.metrics.podMonitor.namespace .Values.namespace (not (empty .Values.metrics.podMonitor.namespace)) }}
+  labels:
+    k8s-app: {{ include "retina.name" . }}
+    {{- if .Values.metrics.podMonitor.additionalLabels }}
+    {{- .Values.metrics.podMonitor.additionalLabels | nindent 4 }}
+    {{- end }}
+spec:
+  podMetricsEndpoints:
+    - port: http
```
Comment (vakalapa) [inline deploy/manifests/controller/helm/retina/templates/podmonitor.yaml:14]: we will have to change this name as well to retina ?

**Review by vakalapa — CHANGES_REQUESTED** [review]
thank you for your contribution, can you add a "testing done" section in description along with some screenshots of the pod monitor if possible. That way we will know if this integration is successful.

File: deploy/manifests/controller/helm/retina/templates/daemonset.yaml (hunk 1 above)
Comment (aslafy-z) [reply deploy/manifests/controller/helm/retina/templates/daemonset.yaml:63]: This port is already scoped to the daemonset pods, I think this change is not necessary. Do you absolutely want me to change this?

Comment (aslafy-z) [inline deploy/manifests/controller/helm/retina/templates/daemonset.yaml:63]:
```suggestion
          - name: retina
```

File: deploy/manifests/controller/helm/retina/templates/daemonset.yaml [hunk 3]
```diff
@@ -148,7 +149,8 @@ spec:
         - name: retinawin
           image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
           ports:
-            - containerPort: {{ .Values.retinaPort }}
+            - name: http
+              containerPort: {{ .Values.retinaPort }}
```
Comment (aslafy-z) [inline deploy/manifests/controller/helm/retina/templates/daemonset.yaml:153]:
```suggestion
          - name: retina
            containerPort: {{ .Values.retinaPort }}
```

File: deploy/manifests/controller/helm/retina/templates/podmonitor.yaml (hunk 2 above)
Comment (aslafy-z) [inline deploy/manifests/controller/helm/retina/templates/podmonitor.yaml:14]:
```suggestion
    - port: retina
```

File: deploy/manifests/controller/helm/retina/templates/daemonset.yaml (hunk 1 above)
Comment (aslafy-z) [reply deploy/manifests/controller/helm/retina/templates/daemonset.yaml]: I made the change.

**Review by rbtr — COMMENTED** [review]
LGTM, thanks for adding this! Can you include evidence of test in the PR description like @vakalapa mentioned? Something like a screenshot of the PodMonitor being discovered in Prometheus would suffice 🙂

**Comment by aslafy-z** [comment]
I updated the description @rbtr @vakalapa

**Review by rbtr — DISMISSED** [review]
thanks @aslafy-z! LGTM

**Comment by aslafy-z** [comment]
I just fixed conflicts @vakalapa @rbtr

**Review by rbtr — APPROVED** [review]

**Review by vakalapa — APPROVED** [review]
thanks for the contribution!

**Pull request merged**